presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/MergePartialAggregationsWithFilter.java

```diff
@@ -436,5 +452,89 @@ public PlanNode visitExchange(ExchangeNode node, RewriteContext<Context> context
+                if (maskExpression == null) {
+                    return new HashSet<>();
+                }
+
+                VariableReferenceExpression variable = extractVariableFromSimpleComparison(maskExpression);
+                if (variable == null) {
+                    return new HashSet<>();
+                }
+
+                commonVariables.add(variable);
+            }
+
+            // If all mask variables reference the same variable in simple comparisons, use that
+            if (commonVariables.size() == 1) {
+                return commonVariables;
+            }
+
+            return new HashSet<>();
+        }
+
+        /**
+         * Gets the expression that defines a mask variable by looking in the source plan.
+         * This is a simple implementation that looks for project nodes.
+         */
+        private RowExpression getMaskVariableExpression(VariableReferenceExpression maskVar)
+        {
+            // For the simple approach, we disable the optimization for now
+            // and always fallback to original behavior. This ensures correctness
+            // while we build the foundation.
+            return null;
+        }
+
+        /**
+         * Extracts the variable from simple comparison expressions like "column = constant".
+         * Returns the variable if this is a simple comparison, null otherwise.
+         */
+        private VariableReferenceExpression extractVariableFromSimpleComparison(RowExpression expression)
+        {
+            if (!(expression instanceof CallExpression)) {
+                return null;
+            }
+
+            CallExpression call = (CallExpression) expression;
+
+            String functionName = call.getDisplayName();
+            if (!isSimpleComparisonOperator(functionName)) {
+                return null;
+            }
+
+            if (call.getArguments().size() != 2) {
+                return null;
+            }
+
+            RowExpression left = call.getArguments().get(0);
+            RowExpression right = call.getArguments().get(1);
+
+            if (left instanceof VariableReferenceExpression && right instanceof ConstantExpression) {
+                return (VariableReferenceExpression) left;
+            }
+
+            if (left instanceof ConstantExpression && right instanceof VariableReferenceExpression) {
+                return (VariableReferenceExpression) right;
+            }
+
+            return null;
+        }
+
+        private boolean isSimpleComparisonOperator(String functionName)
+        {
+            return functionName.equals("$operator$equal") ||
+                    functionName.equals("$operator$not_equal") ||
+                    functionName.equals("$operator$less_than") ||
+                    functionName.equals("$operator$less_than_or_equal") ||
+                    functionName.equals("$operator$greater_than") ||
+                    functionName.equals("$operator$greater_than_or_equal");
+        }
```

Comment:
You might also want to make it recursive - look at the visitor just make it a RowExpressionVisitor